### ISSUE TITLE
[Elastic Agent] Enhance input Health reporting from agent to better convey issues related to installation of `unprivileged` agent 

### ISSUE THREAD

**Issue by nimarezainia** [elastic/elastic-agent#4766]
If the agent is provisioned in the `unprivileged` mode there may be data sources which won't be readable by the agent, as they require higher privilege to be accessed.  This will cause the agent to go into a degraded state and show the integration as unhealthy.

Since the agent knows that it is running in an unprivileged mode AND can recognize that there's an issue with reading the input, it would be great to have this information propagated back to Fleet. Ideally the user has enough information to know that their input is unhealthy due to the fact that agent is in `unprivileged` mode.

cc: @cmacknz would it be possible to add the current log/or health reporting that is sent today and the ideal format for the future where user has ample information to debug.

**Comment by blakerouse** [comment]
This should probably be filed more as a meta issue, with a list of beats or inputs that have actually implemented proper health reporting back to the Elastic Agent. The Elastic Agent itself already has all the mechanisms for this to be a great experience.

1. Runtime protections that allow an input to define that it cannot be ran unless it is root or even non-root. This prevents the input from running and the reason why is reported and propagated back to Fleet Server.
2. Health reporting of an individual input back to Elastic Agent that is then propagated back to Fleet Server. The issue her is that most inputs do not do that at all.

In quick summation, adding health reporting at the input level will provide this information.

**Comment by cmacknz** [comment]
Yes this is more of a Beats/input issue. We may want agent to explicitly tell inputs when agent is running as unprivileged so that they do not have to duplicate the detection logic.

**Comment by ycombinator** [comment]
Transferring to Beats repo per discussion in the issue.
